### ISSUE TITLE
Weak label selection as Training parameter

### ISSUE THREAD

**Issue by Willaem** [crosszamirski/WS-DINO#3]
Hi! I've been trying to reproduce the results shown in your paper for a moment now, and I had gotten subpar performances on NSCB values in all channels (0.30) , which has been infuriating since both the code and cellprofiler pipelines should have been the same. Finally, I found back the parameter that should have been changed during training : the weight assignement for the sampler, which looks like this : 

From WS-DINO_BBBC021.py :
```py
df = pd.read_csv('.../BBBC021_annotated.csv')

idx_list = []
weight_list = []
for i in range(0,38): # 12 for MOA, 38 for compound, 103 for treatment
     idx_list.append(df.index[df.Unique_Compounds == i].tolist())
     length = len(idx_list[i])
     weight_sub_list = [length] * length
     weight_list.append(weight_sub_list)

weight_list_flat = []
for nums in weight_list:
  for val in nums:
    weight_list_flat.append(val)

weighty = 1/torch.tensor(weight_list_flat)
```

I think this code could benefit from being formulated as a function, to avoid global variables, like such : 

```py
def makeWeights(args):
    df = pd.read_csv(f'references/BBBC021_annotated.csv')

    idx_list = []
    weight_list = []

    if args.weak_label_header == 'Unique_MoA':
        for i in range(0,12): # 12 for MOA, 38 for compound, 103 for treatment
             idx_list.append(df.index[df.Unique_MoA == i].tolist())
             length = len(idx_list[i])
             weight_sub_list = [length] * length
             weight_list.append(weight_sub_list)

    if args.weak_label_header == 'Unique_Compounds':
        for i in range(0,38): # 12 for MOA, 38 for compound, 103 for treatment
             idx_list.append(df.index[df.Unique_Compounds == i].tolist())
             length = len(idx_list[i])
             weight_sub_list = [length] * length
             weight_list.append(weight_sub_list)

    if args.weak_label_header == 'Unique_Treatments':
        for i in range(0,103): # 12 for MOA, 38 for compound, 103 for treatment
             idx_list.append(df.index[df.Unique_Treatments == i].tolist())
             length = len(idx_list[i])
             weight_sub_list = [length] * length
             weight_list.append(weight_sub_list)

    weight_list_flat = []
    for nums in weight_list:
      for val in nums:
        weight_list_flat.append(val)

    return 1/torch.tensor(weight_list_flat)
```

Tell me what you think!
